pkg/detector/ospkg/ubuntu/ubuntu.go

```diff
@@ -67,15 +68,37 @@ var (
 	}
 )
 
+type options struct {
+	eolDates map[string]time.Time
+}
+
+type Option func(*options)
+
+// WithEOLDates takes eol dates for testability
+func WithEOLDates(dates map[string]time.Time) Option {
+	return func(opts *options) {
+		opts.eolDates = dates
+	}
+}
+
 // Scanner implements the Ubuntu scanner
 type Scanner struct {
+	*options
```

Comment:
nit: Since there is only one field, defining a new struct may be too much. We may want to add the field directly rather than embedding.

```suggestion
	eolDates map[string]time.Time
```

Reply:
Looks simpler, thanks!
updated in https://github.com/aquasecurity/trivy/pull/8981/commits/dd444352261d8c2c1f453dea97ed516063325a14